### PR TITLE
Avoid Operator while checking commutation relation in DAGDependency

### PR DESCRIPTION
**Pull request by georgios-ts**
### Summary
`_does_commute` takes most of the time while building a `DAGDependency`. There was an overhead due to `Operator.compose`. This PR avoids that by multiplying numpy arrays directly. Essentially it's the same code without validating shapes.

### Details and comments
There is a > 2.5x speed up.
```python
import time
from qiskit.converters import circuit_to_dagdependency
from qiskit.circuit.random import random_circuit


n = 5
for d in [50, 100, 200, 500]:
    qc = random_circuit(n, d, seed=42)
    start = time.time()
    graph = circuit_to_dagdependency(qc)
    stop = time.time()

    print(stop - start)
---
Before: 0.68, 2.69, 7.61, 43.11
After: 0.27, 0.90, 2.67, 15.82
```

### CODE REVIEW

**Review by kdk — COMMENTED** [review]
Thanks @georgios-ts , the performance improvements here look good. Is it possible for this to be rolled back into quantum_info? Maybe as a method on the `Operator` class, or to be added to `predicates` (@chriseclectic )?

File: qiskit/dagcircuit/dagdependency.py [hunk 1]
```diff
@@ -601,9 +601,14 @@ def _unknown_commutator(n):
...
     qarg2 = [qarg.index(q) for q in node2.qargs]
 
-    id_op = Operator(np.eye(2 ** qbit_num))
+    dim = 2 ** qbit_num
+    id_op = np.reshape(np.eye(dim), (2, 2) * qbit_num)
 
-    op12 = id_op.compose(node1.op, qargs=qarg1).compose(node2.op, qargs=qarg2)
-    op21 = id_op.compose(node2.op, qargs=qarg2).compose(node1.op, qargs=qarg1)
+    op1 = np.reshape(node1.op, (2, 2) * len(qarg1))
+    op2 = np.reshape(node2.op, (2, 2) * len(qarg2))
```
Comment (mtreinish) [inline qiskit/dagcircuit/dagdependency.py:608]: Does this work with `node1.op.to_matrix()`? When we did similar tuning in `Optimize1qGatesDecomposition` (in #5909) we found that relying on `__array__` and having numpy create an ndarray implicitly was slower than just constructing the numpy array explicitly and using that for the operation.

Comment (georgios-ts) [reply qiskit/dagcircuit/dagdependency.py]: yes, and it's indeed a bit faster.

**Review by mtreinish — APPROVED** [review]
LGTM. thanks for updating.

**Pull request merged**